app/middlewares/authenticator.js

```diff
@@ -12,8 +13,18 @@ module.exports = (req, res, next) => {
     if (err) {
       return next(errors.unauthenticatedUserError(err));
     }
-    logger.info(`Authenticated! Decoded value: ${decode.payload}`);
-    req.decodedValue = decode.payload;
-    return next();
+
+    const iatDate = new Date(decode.iat * 1000);
+    return models.invalidSessions.findOne({ where: { userId: decode.payload.id } }).then(foundSession => {
+      if (foundSession) {
```

Comment:
`if (foundSession && foundSession.createdAt > iatDate)`

Reply:
Changed in latest commits!